### ISSUE TITLE
createDebounce Type

### ISSUE THREAD

**Issue by MrDesjardins** [davedbase/solid-primitives#68]
Suggestion. The type around `createDebounce` is not user-friendly and could be changed to something more intuitive.

For example, if I have a function with a number as a parameter I cannot do:
```
  const [updateNewWeight, clear] = createDebounce(
    (value:number) => console.log(value),
    200
  );
``` 

Nor can I do:
```
  const [updateNewWeight, clear] = createDebounce<number>(
    (value) => console.log(value),
    200
  );
``` 
Not specifying the type also keep the type to `unknown` which causes an issue if you are debouncing on a strongly typed function:

```
  const [updateNewWeight, clear] = createDebounce(
    (value) => mustHaveNumber(value), // Function expect number
    200
  );
```

**Comment by thetarnav** [comment]
Thanks for the issue, the types shouldn't be hard to fix.
btw, if you're taking about ux, how do you feel about the primitive returning the tuple of [fn, clear] instead of only function like other debounce implementations? I'm asking because I've had an idea to return the clean function as a separate property of fn "object", like this:
```ts
createDebounce = <T extends (...args: unknown[]) => void>(
  func: T,
  wait?: number
): T & { clear: () => void }
```

**Comment by atk** [comment]
That would be an improvement, I guess.

**Comment by MrDesjardins** [comment]
@atk I do not understand. Maybe you can provide an example from the examples I provided?

I tried to type the function:

```
const [updateNewWeight, clear] = createDebounce(
    (value:number) => myFunction(value),
    200
  );
function myFunction(x:number): void { /*Code here*/}
```

As you can see, the first callback function is typed (`value:number`) and the function called is also typed `(x:number)`.

**Comment by thetarnav** [comment]
@MrDesjardins It's not working correctly yet, @atk was just saying thet the issue is indeed easy to fix. Currently being worked on in the pr #69

**Comment by thetarnav** [comment]
Corrected with the the new version: `@1.2.0`
